### PR TITLE
Ghost Role Cancel Fix

### PR DESCRIPTION
**Pull request by Geevies**
* You can now cancel out of joining as a humanoid ghost role by pressing cancel on the species selection screen.

Fixes https://github.com/Aurorastation/Aurora.3/issues/10112

### CODE REVIEW

**Comment by Geevies** [comment]
!bugfix

**Comment by Geevies** [comment]
!review

File: code/controllers/subsystems/ghostroles.dm [hunk 1]
```diff
@@ -151,6 +151,8 @@
 			to_chat(usr, "Unable to spawn: pre_spawn failed. Report this on GitHub")
 			return
 		var/mob/M = S.spawn_mob(usr)
+		if(M == -1)
+			return
```
Comment (NonQueueingMatt) [inline code/controllers/subsystems/ghostroles.dm:155]: not sure I like this type abuse here, I'd just go with 

```var/result = S.spawn_mob(usr)
if(!ismob(result)) return //or whatever checks
else (continue)...
```

**Review by Arrow768 — CHANGES_REQUESTED** [review]
Just remove the cancel option instead?

**Review by Arrow768 — APPROVED** [review]

**Pull request merged**